integrationtest/mapper/pojo-standalone-realbackend/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/standalone/realbackend/schema/management/ElasticsearchSchemaManagerExporterIT.java

```diff
@@ -0,0 +1,246 @@
+import static org.hibernate.search.util.impl.test.JsonHelper.assertJsonEquals;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.hibernate.search.integrationtest.mapper.pojo.standalone.realbackend.testsupport.BackendConfigurations;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.DocumentId;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.FullTextField;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+import org.hibernate.search.mapper.pojo.standalone.mapping.SearchMapping;
+import org.hibernate.search.util.common.SearchException;
+import org.hibernate.search.util.impl.integrationtest.mapper.pojo.standalone.StandalonePojoMappingSetupHelper;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class ElasticsearchSchemaManagerExporterIT {
+
+	@Rule
+	public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+	@Rule
+	public StandalonePojoMappingSetupHelper setupHelper =
+			StandalonePojoMappingSetupHelper.withSingleBackend(
+					MethodHandles.lookup(), BackendConfigurations.simple() );
+	private SearchMapping mapping;
+
+	@Before
+	public void setUp() throws Exception {
+		this.mapping = setupHelper.start()
+				// so that we don't try to do anything with the schema and allow to run without ES being up:
+				.withProperty( "hibernate.search.schema_management.strategy", "none" )
+
+				.withProperty( "hibernate.search.backend.type", "elasticsearch" )
+				.withProperty( "hibernate.search.backend.version_check.enabled", false )
+				.withProperty( "hibernate.search.backend.version", "8.6" )
+
+				.withProperty( "hibernate.search.backends." + Article.BACKEND_NAME + ".type", "elasticsearch" )
+				.withProperty( "hibernate.search.backends." + Article.BACKEND_NAME + ".version_check.enabled", false )
+				.withProperty( "hibernate.search.backends." + Article.BACKEND_NAME + ".version", "8.6" )
+				.setup( Book.class, Article.class );
+	}
+
+	@Test
+	public void elasticsearch() throws IOException {
+		Path directory = temporaryFolder.newFolder().toPath();
+		mapping.scope( Object.class ).schemaManager().exportExpectedSchema( directory );
+
+		assertJsonEquals(
+				"{" +
+						"  \"aliases\": {" +
+						"    \"book-write\": {" +
+						"      \"is_write_index\": true" +
+						"    }," +
+						"    \"book-read\": {" +
+						"      \"is_write_index\": false" +
+						"    }" +
+						"  }," +
+						"  \"mappings\": {" +
+						"    \"properties\": {" +
+						"      \"_entity_type\": {" +
+						"        \"type\": \"keyword\"," +
+						"        \"index\": false," +
+						"        \"doc_values\": true" +
+						"      }" +
+						"    }," +
+						"    \"dynamic\": \"strict\"" +
+						"  }," +
+						"  \"settings\": {}" +
+						"}",
```

Comment:
Same as above:

> We'll need to run those tests against all Elasticsearch/OpenSearch versions. I suspect you won't get the same JSON in all case and thus get failures...
> 
> It might be a good idea to be less strict in your assertions, i.e. use `assertJsonEqualsIgnoringUnknownFields` and only check for a few fields. Also, I'd disable this test for ES 6 and below, because it's going to fail on older versions for sure (because of the concept of named types).

Reply:
hmm right, I thought that that might be the case ... but then this particular test has `withProperty( "hibernate.search.backend.version", "8.6" )` so I'd hopped that that would force the version and keep the json stable? Or you are saying do not force a version but get it form the backend and then pass it in as if there's no ES running ? 😃

Reply:
Ah right, that would work. I missed that.

Though if we only test one version, I'd rather we make sure we test the latest one... I'm absolutely convinced we'll never update this `"8.6"` :)

If you want to test this offline, maybe use `org.hibernate.search.util.impl.integrationtest.backend.elasticsearch.dialect.ElasticsearchTestDialect#getActualVersion`, and disable the test for older versions of Elasticsearch?

Reply:
ok will do 😃